File: scala-spark/src/main/scala/com/myCompany/processing/Transformation.scala
package com.myCompany.processing

import com.myCompany.Launch.pathToDataFile
import org.apache.spark.sql.SparkSession

object Transformation {
  def wordCount(spark: SparkSession): Unit ={
    // Read File from local filesystem and create an RDD
    val data = spark.read.textFile(pathToDataFile).rdd
    // split words against word delimiters
    val elements = data.flatMap(l => l.split("[;\\s]"))
    // map operation to index words
    val paires = elements.map( element => (element, 1))
    //reduce operation to count word
    val counts = paires.reduceByKey((a,b) => a+b)
    // display results
    counts.foreach(println)
  }
}

File: java-basics/src/main/java/com/myCompany/processing/Transformation.java
package com.myCompany.processing;

import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;

import java.util.Arrays;

import static com.myCompany.Launch.pathToDataFile;

public class Transformation {
    public static void wordCount(JavaSparkContext jsc) {
        // Read File from local filesystem and create an RDD
        JavaRDD<String> data = jsc.textFile(pathToDataFile);
        // split words against word delimiters
        JavaRDD<String> elements = data.flatMap(content -> Arrays.asList(content.split("[;\\s]")).iterator());
        // map operation to index words
        JavaPairRDD<String, Integer> paires = elements.mapToPair(element -> new Tuple2(element, 1));
        //reduce operation to count word
        JavaPairRDD<String, Integer> counts = paires.reduceByKey((a,b) -> a+b);
        // display results
        counts.foreach(l -> System.out.println(l));
    }
}
